### ISSUE TITLE
(only pedestrian) routing lost - how to recover or is it a bug?

### ISSUE THREAD

**Issue by itsbrody** [cgeo/cgeo#14759]
### Describe your problem!

I'm often walking for caches, that means I use the pedestrian routing profile and this worked very well.
As preparation for a "far from home trip" I downloaded map and routing information to do the same abroad (cache walks in the city). But when I tried to route "pedestrian" in the city this did not work. Also updating the routing information did not help. I thought that all routing capability has lost, although all settings seem ok - it also does not differ between cgeo internal and brouter routing.

When now back in my home location I accidently switched to "bike routing" and surprisingly that works - the same as car routing. Only "pedestrian routing" is still lost and I don't know, how to recover or reset it.

### How to reproduce?

select a cache and switch in fast settings between the different routing capabilities.

none, bike, car are working
pedestrian does same as none.

### Actual result after these steps?

pedistrian  shows a straight line between my position and the selected cache. Bike, Car a routed line to the cache

### Expected result after these steps?

also ped. routing should show a "route".

### Reproducible

Yes

### c:geo Version

2023.09.26

### System information

```text
d.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:81, dirs:0, totalFileSize:110,8 KB, free space: 4,3 GB, files on device: 3194880)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,3 GB, files on device: 3194880)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:13, dirs:1, totalFileSize:644,0 MB, free space: 4,3 GB, files on device: 3194880)
  - ROUTING_TILES: disk/Kartenmaterial/brouter (User-Defined)[disk/Kartenmaterial/brouter[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fbrouter::]] (Uri: content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fbrouter/document/3337-3961%3AKartenmaterial%2Fbrouter, Av:true, files:8, dirs:0, totalFileSize:776,2 MB, free space: 42,5 GB, files on device: 1331257200)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,3 GB, files on device: -1)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #5
  - content://com.android.externalstorage.documents/tree/primary%3ADownload%2Fapplications%2Fcgeo%2Fgpx (10. Juli 2022, 15:14):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (25. Juni 2022, 12:52):RW
  - content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fthemes (25. Juni 2022, 14:14):RW
  - content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fosm (25. Juni 2022, 14:14):RW
  - content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fbrouter (27. Aug., 08:32):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:82,7 MB) on system internal storage

--- End of system information ---
```

**Comment by itsbrody** [comment]
just saw that I partially copied the system information - here is the rest

wolfgang, [25.10.2023 18:36]
## System information

c:geo version: 2023.09.26

Device:
-------
- Device type: FP3 (FP3, Fairphone)
- Available processors: 8
- Android version: 11
- Android build: 8901.4.A.0023.0 release-keys
- Screen resolution: 1080x2034px (411x750dp)
- Pixel density: 3.01875
- System font scale: 1.15 / used scale: 1.15
- Sailfish OS detected: false
- Google Play services: enabled - 23.11.15 (150400-520109968)
- Memory:  Available:1,2 GB, Total:3,5 GB, Threshold: 216,0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: orientation

Program settings:
-------
- Settings: v9, Count:217
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+02:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: 24. März 2020, 20:42
- Last auto backup: 20. Okt., 12:00
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Kombiniert (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMultiMapSource
  - Atts: Alps+west (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Austria.map: / Map data (c) OpenStreetMap contributors /  / Bgr+ (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Bulgaria (OAM).map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / bulgaria.map: / Map data (c) OpenStreetMap contributors /  / Che+ (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Cyp (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Cyprus (OAM).map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2023 /  / Deu+sued (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Deu+west (FZK).map: / (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti /  / Georgia (OAM).map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2023
  - Theme: fzk-outdoor-soft-v5.zip:fzk-outdoor-soft-v5/fzk-outdoor-soft-v5.xml

Filters:
-------
- Hide waypoints: -
- LIVE: Status: Gefunden=Nein ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status:found_no))
- OFFLINE: Keine ([:inconclusive=false:advanced=true])

Stored Filters (#3):
- Motorrad: [Motorrad:inconclusive=false:advanced=true]AND(name:MO:starts_with;type:WEBCAM:MULTI:TRADITIONAL;difficulty_terrain:d=-:d=-:t=-:t=2.5;status:owned_no:exclude_archived))
- TB Hotels (auf Fahrten): [TB Hotels \(auf Fahrten\):inconclusive=false:advanced=false]AND(name:hotel:contains;type:MULTI:LETTERBOX:TRADITIONAL;difficulty_terrain:d=-:d=-:t=1.0:t=3.0;status:owned_no:exclude_disabled:exclude_archived))
- TB Hotels (auf Fahrten) mit TB's: [TB Hotels \(auf Fahrten\) mit TB's:inconclusive=false:advanced=false]AND(name:hotel:contains;type:MULTI:LETTERBOX:TRADITIONAL;difficulty_terrain:d=-:d=-:t=1.0:t=3.0;status:owned_no:has_trackable_yes:exclude_disabled:exclude_archived))

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   - Geocaching.com Adventure Lab
   - opencaching.de: Logged in (Anmeldung OK)
- Geocaching.com date format: dd MMM yy
- Geocaching.com

wolfgang, [25.10.2023 18:36]
website language: en-US
- Last login error on geocaching.com: communication error (2023-10-01 10-34)
- Last successful login on geocaching.com: 2023-10-25 18-35
- Routing: internal / BRouter installed: true
- Installed c:geo plugins:  none

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: DENIED
- READ_EXTERNAL_STORAGE: granted
- WRITE_EXTERNAL_STORAGE: DENIED
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (4,3 GB free) v3 internal isDir(10 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (4,3 GB free) v3 internal isDir(10 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (4,3 GB free) v3 external non-removable isDir(1431 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (4,3 GB free) v3 internal isDir(0 entries)
- Map render theme path: fzk-outdoor-soft-v5.zip:fzk-outdoor-soft-v5/fzk-outdoor-soft-v5.xml
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=6, dirs:>=14, totalFileSize:>=164,0 MB, free space: 4,3 GB, files on device: 3194880)
  - OFFLINE_MAPS: disk/Kartenmaterial/osm (User-Defined)[disk/Kartenmaterial/osm[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fosm::]] (Uri: content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fosm/document/3337-3961%3AKartenmaterial%2Fosm, Av:true, files:29, dirs:3, totalFileSize:12,0 GB, free space: 42,5 GB, files on device: 1331257200)
  - OFFLINE_MAP_THEMES: disk/Kartenmaterial/themes (User-Defined)[disk/Kartenmaterial/themes[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fthemes::]] (Uri: content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fthemes/document/3337-3961%3AKartenmaterial%2Fthemes, Av:true, files:13, dirs:0, totalFileSize:2,3 MB, free space: 42,5 GB, files on device: 1331257200)
  - OFFLINE_MAP_SHADING: disk/Kartenmaterial/osm/_hgt (Default)[disk/Kartenmaterial/osm/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fosm::/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/3337-3961%3AKartenmaterial%2Fosm/document/3337-3961%3AKartenmaterial%2Fosm%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 42,5 GB, files on device: 1331257200)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,3 GB, files on device: 3194880)
  - GPX: /Download/applications/cgeo/gpx (User-Defined)[/Download/applications/cgeo/gpx[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ADownload%2Fapplications%2Fcgeo%2Fgpx::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ADownload%2Fapplications%2Fcgeo%2Fgpx/document/primary%3ADownload%2Fapplications%2Fcgeo%2Fgpx, Av:true, files:65, dirs:0, totalFileSize:13,1 MB, free space: 4,3 GB, files on device: 3194880)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=8, dirs:>=7, totalFileSize:>=246,2 MB, free space: 4,3 GB, files on device: 3194880)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.androi

**Comment by itsbrody** [comment]
"cross-check in Einstellungen => Routing / Navigation => "Laufen"-Profil"

Yes and all the other Profile possibilities (Fahrrad, Auto) are also equipped with a profile name.
While the latter both work, pedestrian does (no more) work.

After some more checks:
it seems to be a new "directory" error at all, with "internal routing" activated using a dedicated externalized directory for routing files. This configration has not changed all the time. That means, that the profile files are all and only in cgeo/routing. Only the "big" routing data files have been externalized all the time.

Hth

**Comment by Lineflyer** [comment]
Did you try different starting points or only a route between two specific locations in that area?

If there is an available bike route its most likely, that also a pedestrian route should be able to be routed. I did however once experience, that there is no possible walk path to my target, which leads to no route being shown. 
In my case it was a problem of the underlying path type info, which declared a walk path as highway, which is therefore denied to be used for pedestrian routing.

**Comment by rouroni1** [comment]
Hi.

I had this issue with the latest version of C:Geo on OpenAndroMaps with Elevation Theme (Android 13 - Poco F4 GT)
With Hiking routing worked, but Car didn't -> Geocache https://coord.info/GCAFDHV .
This geocache is 1 or 2 meters off the road.

What I did that solved the Issue.
I opened geocache https://coord.info/GC6W2DV with routing for Car selected, and the route was generated.

Since then the issue is gone.

**Comment by itsbrody** [comment]
> Did you try different starting points or only a route between two specific locations in that area?

I tried several times, when I was abroad and in a big city. Just yesterday I tried again, where the routing before "the event which destroyed pedestrian routing" did very well. Checking the routing data says "they are up to date". All seems to be setup correctly (all the time), but since then, it does not work...

 
 

Gesendet: Donnerstag, 26. Oktober 2023 um 19:03 Uhr
Von: "Lineflyer" ***@***.***>
An: "cgeo/cgeo" ***@***.***>
Cc: "Wolfgang" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [cgeo/cgeo] (only pedestrian) routing lost - how to recover or is it a bug? (Issue #14759)


 

Did you try different starting points or only a route between two specific locations in that area?

If there is an available bike route its most likely, that also a pedestrian route should be able to be routed. I did however once experience, that there is no possible walk path to my target, which leads to no route being shown.
In my case it was a problem of the underlying path type info, which declared a walk path as highway, which is therefore denied to be used for pedestrian routing.

**Comment by itsbrody** [comment]
Checked it again - with upTOdate APP (on my phone) and as the APP say "no updates of routing data".


settings under "Routing / Navigation" (with Extended Settings enabled)
Routing / Navigation

"internal Routing" tagged

"Drawer with BRouter Routingdata" (as before) externalized on memory card; when selecting this entry a popup grows saying that therin are 8 files with ...MB"
Walking-Profile "trekking.brf"

Biking-Profile "fastbike.brf"

Car-Profile "car-eco.brf"

 

 

car routing is working

bike routing is working


walk still does not work

 

when deselection "internal routing" all the mentioned settings go "deselected"

 

 

 

Gesendet: Dienstag, 31. Oktober 2023 um 11:51 Uhr
Von: "Wolfgang Brodowski" ***@***.***>
***@***.***
Betreff: Aw: Re: [cgeo/cgeo] (only pedestrian) routing lost - how to recover or is it a bug? (Issue #14759)



> Did you try different starting points or only a route between two specific locations in that area?

I tried several times, when I was abroad and in a big city. Just yesterday I tried again, where the routing before "the event which destroyed pedestrian routing" did very well. Checking the routing data says "they are up to date". All seems to be setup correctly (all the time), but since then, it does not work...

 
 

Gesendet: Donnerstag, 26. Oktober 2023 um 19:03 Uhr
Von: "Lineflyer" ***@***.***>
An: "cgeo/cgeo" ***@***.***>
Cc: "Wolfgang" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [cgeo/cgeo] (only pedestrian) routing lost - how to recover or is it a bug? (Issue #14759)


 

Did you try different starting points or only a route between two specific locations in that area?

If there is an available bike route its most likely, that also a pedestrian route should be able to be routed. I did however once experience, that there is no possible walk path to my target, which leads to no route being shown.
In my case it was a problem of the underlying path type info, which declared a walk path as highway, which is therefore denied to be used for pedestrian routing.

**Comment by moving-bits** [comment]
What happens if you open your file manager, navigate to the folder where "trekking.brf" is stored (normally /cgeo/routing/), and delete the file "trekking.brf"?
c:geo will recreate this file with the integrated default for "trekking.brf" when you start the next route with this profile.

**Comment by itsbrody** [comment]
a) The regeneration of "trekking.brf" worked; it made the file bigger from ~12.3k to 15.3k

b) some configuration tests

b1) with (pedestrian / bike / car) --> {trekking, fastbike, car-eco} :: (again) bike, car -- OK; pedestrian -- NOK

b2) with (pedestrian / bike / car) --> {fastbike, fastbike, fastbike} :: bike -- OK, pedestrian, car -- NOK

b3}with (pedestrian / bike / car) --> {car-eco, car-eco, car-fast} :: bike, car -- OK; pedestrian -- NOK

b4.1} as b3) added with userdefined 1 and 2  as {mopped, trekking} :: first work, second don't work

b4.2} as b3) added with userdefined 1 and 2  as {trekking,mopped} :: first don't work, second work


 

BTW: How can I deselect userdefined 1 and 2 again?

 

Gesendet: Mittwoch, 15. November 2023 um 11:20 Uhr
Von: "moving-bits" ***@***.***>
An: "cgeo/cgeo" ***@***.***>
Cc: "Wolfgang" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [cgeo/cgeo] (only pedestrian) routing lost - how to recover or is it a bug? (Issue #14759)


 

What happens if you open your file manager, navigate to the folder where "trekking.brf" is stored (normally /cgeo/routing/), and delete the file "trekking.brf"?
c:geo will recreate this file with the integrated default for "trekking.brf" when you start the next route with this profile.

**Comment by itsbrody** [comment]
In the meantime I'm on V2024.01.06 - I've checked anything. Have put the *.brf-files from another and working installation. Have further reconfigured routing files again to internal memory. But nothing help.

Last and now I configured every of the possible rounting buttons to the same .brf-file (e.g. car-eco). And on the map I selected a cache near a streat and near enough to my current location. Result: Selection any other of the real routing profile (besides pedestrian) worked and gave me the same route. Pedestrian made a line only

**Comment by moving-bits** [comment]
seems that somehow Android on that specific mobile has put some restriction on `trekking.brf` in that folder, so that c:geo cannot access it. I don't know how to recover from that lock.

Can you copy a working `trekking.brf` (from a different installation) into that folder, but using a different file name, and then select this one for the `walk` profile?

**Comment by itsbrody** [comment]
Thanks for suggesting this, but the name isn't relevant.

From further testing with following configuration (see attachment 1.png), I in the meantime guess that red circled buttons (see attachment 2.png) do the identical job. You also see there on selection 2 is show a directed and a profile  specific distance(green underlined), whilst selecting „pedestrian“ only shows a direct distance.

**Comment by moving-bits** [comment]
> Thanks for suggesting this, but the name isn't relevant.

What I meant is:
- Copy an existing `trekking.brf` on a device where routing mode "walk" does work, and give it a different name, other than `trekking.brf`. 
- Then copy that file under its new name to the device where rounting mode "walk" does not work, and store it in the routing folder, NOT overwriting the existing `trekking.brf`. 
- Then select the new file as profile for "walk", and see it that works. 
- (So both file will coexist then, both the non-working `trekking.brf` and a new one under a different name.)

> From further testing with following configuration (see attachment 1.png), I in the meantime guess that red circled buttons (see attachment 2.png) do the identical job. You also see there on selection 2 is show a directed and a profile  specific distance(green underlined), whilst selecting „pedestrian“ only shows a direct distance.

Your attachments did not reach the GitHub page as you sent them via email. Can you open the website and attach your screenshots there, please?

**Comment by itsbrody** [comment]
> Your attachments did not reach the GitHub page as you sent them via email. Can you open the website and attach your screenshots there, please?

here they are:
![1](https://github.com/cgeo/cgeo/assets/19346144/5bb122f4-a7aa-45f6-ac2c-e5f7b7dd2af9)
![2](https://github.com/cgeo/cgeo/assets/19346144/8c96b175-12ee-469d-92c1-f4eeb4b4f313)

**Comment by itsbrody** [comment]
> > Thanks for suggesting this, but the name isn't relevant.
> 
> What I meant is:
[...]
>     * (So both file will coexist then, both the non-working `trekking.brf` and a new one under a different name.)

I transferred the complete routing directory (as ZIP archive) from and exchanged it with the current (by renaming the current to _routing.old_). Then I dearchived the ZIP to the freed place. I copied trekking to my DL directory, renamed it to zufuss.brf and copied it back.

Results are shown above. Does this also fit  from your point of view?

**Comment by itsbrody** [comment]
BTW - yesterday I again walked along some caches, explicitely using [2] with good routed access to the pathes, I walked on.

**Comment by moving-bits** [comment]
Thanks for adding the pics. I still do not understand why the "walk" profile does not work on that specific device. As we have no other reports for this and it does work on your second device, it seems to be something device-specific. But at least there is a workaround with "zufuss.brf" using profile "2". But I'm still curious what's happening here...

@eddiemuc 
Is there some logging we could activate to track this down?

**Comment by itsbrody** [comment]
I'm using the concerned device since round about four years. I've gotten several main and minor updates from the android OS.

I've first discovered the problem when far away from home last summer/autumn. The last OS major upgrade were a lot of time before, since then only minors.

The idea to check the other profiles including the extra profiles, I found some time - don't know when they came in, but my other installation does not have them - so might be the adding extra profiles made a mistake, which gave that behaviour? Or what have been changed on "Routing" before I found this error and not discovered from others. MIght also be that they are using bikes, cars but do not walk the lot.

 

 

 
 

Gesendet: Montag, 19. Februar 2024 um 16:25 Uhr
Von: "moving-bits" ***@***.***>
An: "cgeo/cgeo" ***@***.***>
Cc: "Wolfgang" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [cgeo/cgeo] (only pedestrian) routing lost - how to recover or is it a bug? (Issue #14759)


 

Thanks for adding the pics. I still do not understand why the "walk" profile does not work on that specific device. As we have no other reports for this and it does work on your second device, it seems to be something device-specific. But at least there is a workaround with "zufuss.brf" using profile "2". But I'm still curious what's happening here...

@eddiemuc
Is there some logging we could activate to track this down?

**Comment by eddiemuc** [comment]
> @eddiemuc 
> Is there some logging we could activate to track this down?

No idea. Turning on debug mode could not hurt

**Comment by moving-bits** [comment]
> > @eddiemuc
> > Is there some logging we could activate to track this down?
> 
> No idea. Turning on debug mode could not hurt

Is there some way to get a list of files/folders SAF is holding kind of a lock on? Or some way to remove such locks?

**Comment by eddiemuc** [comment]
> > > @eddiemuc
> > > Is there some logging we could activate to track this down?
> > 
> > No idea. Turning on debug mode could not hurt
> 
> Is there some way to get a list of files/folders SAF is holding kind of a lock on? Or some way to remove such locks?

That would be the info in the system information. SAF holds permissions to access either single files or whole folders (and it's subfolders). We mostly use the later.

**Comment by itsbrody** [comment]
> > Is there some way to get a list of files/folders SAF is holding kind of a lock on? Or some way to remove such locks?
> 
> That would be the info in the system information. SAF holds permissions to access either single files or whole folders (and it's subfolders). We mostly use the later.

Don't knowing what SAF is.

IMHO a lock would also influence using the identical .brf-file on any other of that icons, but either that usecase (using "zufuss.brf" on pedestrian NOK and on [2] is OK) or using another working .brf-file (e.g. fastbike.brf) on origin "bike" OK and "pedestrian" NOK contradicts this theory. Further a lock should be released, when either the APP is quit or latest the OS is shutdown and restarted.
Anything I oversaw?

**Comment by itsbrody** [comment]
> > Don't knowing what SAF is.
> 
> SAF = Storage Access Framework. Android's wonderful framework to make otherwise easy file access a complex and painful thing. See e.g. https://developer.android.com/guide/topics/providers/document-provider

Thanks for clarifying that abbreviation.
> 
> > IMHO a lock would also influence ...
> 
> SAF does not lock files and such. It deals with permissions to read/write files and folders. It does not lock them.

This does concern [this questions](https://github.com/cgeo/cgeo/issues/14759#issuecomment-1953814846) from @moving-bits

**Comment by moving-bits** [comment]
> > Is there some way to get a list of files/folders SAF is holding kind of a lock on? Or some way to remove such locks?
> 
> That would be the info in the system information. SAF holds permissions to access either single files or whole folders (and it's subfolders). We mostly use the later.

Thanks, what I meant is the other way around - Android preventing a file to be visible to c:geo. But I'm not sure whether it's actually a SAF problem.

@itsbrody 
Long thread, meanwhile, and not sure if I have the working and the non-working settings clearly separated anymore. Let me try to summarize:
- original "shortest.brf" does not work in WALK profile, but does work when assigned to one of the other profiles?
- using a copied version (from a different device) and assigned to WALK profile still does not work (which could exclude one of the alleged SAF-"hiccups"), but does work for other profiles?
- assigning a different template (eg.: trekking.brf) for WALK profile still does not work?
- "does not work" meaning: a straight line being drawn between current position and target (when using WALK profile).
- Whereas on a different device, everything works as expected.

Does that summarize it correctly?

You could two things for further analysis:
- Create a logfile from within c:geo, then try routing in WALK profile, create yet another logfile (detailed version) and send the second one to our support address [support - at - cgeo.org]. (Creating the first one is just for clearing the buffer.)
- Create a backup on both devices, and sent the two xml files (not the actual databases, .sqlite) to support, so that we can compare every setting. Backup do NOT need o include sensitive data, you can uncheck this when creating the backups.

Thanks.

**Comment by itsbrody** [comment]
> Does that summarize it correctly?

correct - besides the last point in that I did not put back that files to the other device, where they came from.

Additional want to mention that "other device" is running Android9, whilst here we're on Android11


> You could two things for further analysis:
> 
>     * Create a logfile from within c:geo, then try routing in WALK profile, create yet another logfile (detailed version) and send the second one to our support address [support - at - cgeo.org]. (Creating the first one is just for clearing the buffer.)
> 
>     * Create a backup on both devices, and sent the two xml files (not the actual databases, .sqlite) to support, so that we can compare every setting. Backup do NOT need o include sensitive data, you can uncheck this when creating the backups.
> 
> 
> Thanks.

will do soon. Is creating such detailed log file  and backup self-explaining in c:geo?

**Comment by eddiemuc** [comment]
> Thanks, what I meant is the other way around - Android preventing a file to be visible to c:geo. But I'm not sure whether it's actually a SAF problem.

We can't make visible files which we don't see. Because we don't see them :-)
What we can do of course is to log all the files we can see (e.g. in a specific folder). This was actually implemented for system information in earlier versions when SAF was fresh. We removed it since for privacy reasons and just display file counts.

What one could do is check in sys info whether the displayed file count for routing folder matches the number of files. If sys I do displays less files than really available then apparently we can't see all of them

**Comment by itsbrody** [comment]
> > Thanks, what I meant is the other way around - Android preventing a file to be visible to c:geo. But I'm not sure whether it's actually a SAF problem.
> 
> We can't make visible files which we don't see. Because we don't see them :-) 

Generally I understand, but why do you (potentially) see same file, when linked to "[2]" or "car" or .... but not linked to "walking"?

**Comment by itsbrody** [comment]
> You could two things for further analysis:
> 
>     * Create a logfile from within c:geo, then try routing in WALK profile, create yet another logfile (detailed version) and send the second one to our support address [support - at - cgeo.org]. (Creating the first one is just for clearing the buffer.)
> 
>     * Create a backup on both devices, and sent the two xml files (not the actual databases, .sqlite) to support, so that we can compare every setting. Backup do NOT need o include sensitive data, you can uncheck this when creating the backups.
> 
> 
> Thanks.

Done and send.

**Comment by eddiemuc** [comment]
> > > Thanks, what I meant is the other way around - Android preventing a file to be visible to c:geo. But I'm not sure whether it's actually a SAF problem.
> > 
> > We can't make visible files which we don't see. Because we don't see them :-) 
> 
> Generally I understand, but why do you (potentially) see same file, when linked to "[2]" or "car" or .... but not linked to "walking"?

Sorry for the confusion, I was just trying to clarify the options to debug into the problem.

Unfortunately I have currently no idea why we should not be able to read some files in a folder and not be able to read others.

**Comment by moving-bits** [comment]
> > Thanks, what I meant is the other way around - Android preventing a file to be visible to c:geo. But I'm not sure whether it's actually a SAF problem.
> 
> We can't make visible files which we don't see. Because we don't see them :-) 

My hope was that there is some Android mechanism (maybe dev options) to list such exclusions.

> What one could do is check in sys info whether the displayed file count for routing folder matches the number of files. If sys I do displays less files than really available then apparently we can't see all of them

@itsbrody 
This is another thing that you could do (as time permits): 
- Count the number of files in your configured "routing" folder (the one which contains the ".brf" files)
- and compare that number to the one shown in system info in the "Paths" - "ROUTING_BASE" section.

System info can be found by
- going to home screen
- tap on the c:geo icon in the upper left
- and swipe to the tab named "system"

**Comment by eddiemuc** [comment]
> The logs show one error from the internal routing engine, that it cannot read a binary cache file:
> 
> ```
> 02-21 11:13:08.325 29780 29799 I cgeo : [Binder:29780_4] java.lang.RuntimeException: Exception reading rawTrack: java.io.EOFException
> 02-21 11:13:08.325 29780 29799 I cgeo : at cgeo.geocaching.brouter.core.OsmTrack.readBinary(OsmTrack.java:131)
...
> ```
> 
> Reason for this is yet unclear.

first guess is that the file ended prematurely (eg end of file was reached unexpectedly while reading some data). This would happen if the file was corrupted e.g. "cut off" during a failed cop or download. 

But I believe you already ruled out file corruption before?

**Comment by eddiemuc** [comment]
Addendums:
- error seems to happen while reading a binary file (to find a track). This leads to the question if maybe one of the segment files is corrupt (not the profile files)?
- in class OsmTrack.java, line 131, the original exception gets lost because it is just printed instead of passing it as cause. This makes it hard to identify original error cause. Should be changed in code for the future
- looking into router code this somehow seems to be related to routing to "not direct" waypoints. Don't know what this term means - maybe error only happens for target waypoints which are offside any road?

**Comment by devemux86** [comment]
> Exception reading rawTrack

I have already reported it in BRouter:
- https://github.com/abrensch/brouter/issues/671

I also sent the fix and it was merged:
- https://github.com/abrensch/brouter/pull/672

**Comment by itsbrody** [comment]
answering my tasks (by mail):

c:geo says:

ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:11, dirs:1, totalFileSize:214,9 MB, free space: 2,6 GB, files on device: 3194880)


Numbers in dir (as shown by file explorer : 11 entries (separated: 10 files, 1 dir).
The ten files are nine .brf and the lookups.dat


Also showing hidden files does not offer more files

**Comment by moving-bits** [comment]
> > Exception reading rawTrack
> 
> I have already reported it in BRouter:
> I also sent the fix and it was merged:

Thanks for the pointer. I've added this fix to c:geo (#15366).

**Comment by moving-bits** [comment]
fixed with #15366, so let's close here
